src/com/rockbite/bootcamp/ShopImpl.java

```diff
@@ -0,0 +1,177 @@
+package com.rockbite.bootcamp;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * implementation of the IShop interface, represents shop API
+ * in current version it have to be a singleton to make easy to implement command pattern on it
+ */
+public class ShopImpl implements IShop {
+
+    private static ShopImpl instance;
+
+    //items taken from buyers
+    private Map<Integer, Item> itemsInCashDesk;
+
+    //products to be sold in the shop
+    private List<Product> products;
+
+    private Map<Integer, Product> soldProducts;
```

Comment:
why would shop remember it's sold products?
It seems like you are trying to do it under a purchase but NOT with using proper command pattern.